PreLoad.cmake

```diff
@@ -58,10 +58,9 @@ execute_process(COMMAND uname -m OUTPUT_VARIABLE ARCH)
 # By default, prefer clang on Linux
 # But note, that you still may change the compiler with -DCMAKE_C_COMPILER/-DCMAKE_CXX_COMPILER.
 if (OS MATCHES "Linux"
+    # some build systems may use CC/CXX env variables
     AND "$ENV{CC}" STREQUAL ""
-    AND "$ENV{CXX}" STREQUAL ""
-    AND NOT DEFINED CMAKE_C_COMPILER
-    AND NOT DEFINED CMAKE_CXX_COMPILER)
+    AND "$ENV{CXX}" STREQUAL "")
```

Comment:
Maybe I missed it but how is the removal of l. 63/64 (left side) related to s390/x?